client/db/bolt/db.go

```diff
@@ -719,12 +768,49 @@ func decodeOrderBucket(oid []byte, oBkt *bbolt.Bucket) (*dexdb.MetaOrder, error)
 	}, nil
 }
 
+// updateOrderBucket expects an oid to exist in either the orders or archived
+// order buckets. If status is greater than booked, it first checks active
+// orders. If found it moves the order to the archived bucket and returns the
+// order bucket. If status is less than booked, it expects the order to already
+// be in active orders and returns the bucket.
+func updateOrderBucket(ob, aob *bbolt.Bucket, oid order.OrderID, status order.OrderStatus) (*bbolt.Bucket, error) {
+	executed := status > order.OrderStatusBooked
+	if executed {
+		if bkt := ob.Bucket(oid[:]); bkt != nil {
+			oBkt, err := aob.CreateBucket(oid[:])
+			if err != nil {
+				return nil, err
+			}
+			// NOTE: Assumes no buckets saved in bkt.
+			bkt.ForEach(func(k, v []byte) error {
+				return oBkt.Put(k, v)
+			})
+			ob.DeleteBucket(oid[:])
+			return oBkt, nil
+		}
+		oBkt := aob.Bucket(oid[:])
+		if oBkt == nil {
+			return nil, fmt.Errorf("order %s not found", oid)
+		}
+		return oBkt, nil
+	}
+	if aob.Bucket(oid[:]) != nil {
+		return nil, fmt.Errorf("archived order %s set to active", oid)
+	}
```

Comment:
Maybe this check is not necessary?

Reply:
Same thought in the other function.  It's also one of the expensive things we are trying to avoid.